src/entities/simulation/simulation-scenario-config/subscription-cancellation-details.ts

```diff
@@ -0,0 +1,13 @@
+import { type SimulationSubscriptionCancellationConfig } from '../../../types/index.js';
+import { SubscriptionCancellationEntities } from './subscription-cancellation-entities.js';
+import { SubscriptionCancellationOptions } from './subscription-cancellation-options.js';
+
+export class SubscriptionCancellationDetails {
+  public readonly entities?: SubscriptionCancellationEntities | undefined;
+  public readonly options?: SubscriptionCancellationOptions | undefined;
+
+  constructor(config: SimulationSubscriptionCancellationConfig) {
+    this.entities = config.entities ? new SubscriptionCancellationEntities(config.entities) : undefined;
+    this.options = config.options ? new SubscriptionCancellationOptions(config.options) : undefined;
```

Comment:
```suggestion
  public readonly entities?: SubscriptionCancellationEntities | null;
  public readonly options?: SubscriptionCancellationOptions | null;

  constructor(config: SimulationSubscriptionCancellationConfig) {
    this.entities = config.entities ? new SubscriptionCancellationEntities(config.entities) : null;
    this.options = config.options ? new SubscriptionCancellationOptions(config.options) : null;
```

I don't believe we explicitly set `undefined` anywhere else. Please do the same change in other classes too